server/handlers.go

```diff
@@ -110,6 +109,9 @@ func (s *Server) discoveryHandler() (http.HandlerFunc, error) {
 	}
 	sort.Strings(d.ResponseTypes)
 
+	d.GrantTypes = s.supportedGrantTypes
+	sort.Strings(d.GrantTypes)
```

Comment:
Since this modifies the original slice, I think the sorting should rather be done where the slice is created. Consumers should treat it as read-only.

Also, the sorting called in the handler will be done every time the discovery endpoint is called.

Any reason why it should be kept here?

Reply:
@sagikazarmark change made! 

It was here only imitate the treatment of  `responseTypes`

Reply:
Hm, yeah, I think that shouldn't be there either. But let's not block this anymore.